src/Process/Coverage/CoverageChecker.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Process\Coverage;
+
+use Composer\XdebugHandler\XdebugHandler;
+
+/**
+ * @internal
+ */
+final class CoverageChecker
```

Comment:
tbh it's not a coverage checker :) should have a bit different name but can't give you any variant right now

Reply:
Yeah youre right, but i can't think of a better name either.

It checks if it is provided with coverage, or if it can generate coverage (it has a debugger), so it was teh best i could come up with.

Reply:
Coverage Existence Checker?

Reply:
I don't think thats good either, as the coverage doesn't have to exist, as it is also okay if we can create it during our run.